internal/artifact/azure_blob.go

```diff
@@ -0,0 +1,80 @@
+package artifact
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+	"github.com/buildkite/agent/v3/logger"
+)
+
+// The domain suffix for Azure Blob storage.
+const azureBlobHostSuffix = ".blob.core.windows.net"
+
+// NewAzureBlobClient creates a new Azure Blob client.
+func NewAzureBlobClient(l logger.Logger, storageAccountName string) (*service.Client, error) {
+	url := fmt.Sprintf("https://%s%s/", storageAccountName, azureBlobHostSuffix)
+
+	// TODO: other credentials?
```

Comment:
Did you do these? If not, maybe provide some links to docs on what the remaining credentials are.